backend/src/test/java/com/github/neuefische/backend/service/RestaurantServiceTest.java

```diff
@@ -26,4 +28,23 @@ void getAllRestaurants_whenCalledInitially_ThenReturnEmptyList() {
         assertEquals(expected, actual);
         verify(repo).findAll();
     }
+
+    @Test
+    void getRestaurantById_whenCalledWithInvalidId_thenThrowNoSuchElementException(){
+        assertThrows(NoSuchElementException.class, () -> {service.getRestaurantById("123");});
+    }
+
+    @Test
+    void getRestaurantById_whenCalledWithValidId_thenRestaurantWithId(){
+        //GIVEN
+        Restaurant expected = new Restaurant("1","Okinii", "Cologne" );
+        when(repo.findById("1")).thenReturn(Optional.of(expected));
+
+        //WHEN
+        Restaurant actual = service.getRestaurantById("1");
+
+        //THEN
+        assertEquals(expected,actual);
+        verify(repo).findById("1");
```

Comment:
```suggestion
        verify(repo).findById("1");
        assertEquals(expected,actual);
```

verify should come first, because after an assertion the method is finished